tests/defences/detector/poison/test_activation_defence.py

```diff
@@ -28,285 +28,290 @@
-        )
-        _, is_clean_lst_gen = self.defence_gen.detect_poison(
-            nb_clusters=3, nb_dims=10, reduce="PCA", cluster_analysis="distance"
-        )
-        self.assertEqual(len(x_train), len(is_clean_lst))
-        self.assertEqual(len(x_train), len(is_clean_lst_gen))
-
-        # Test change of state to new number of clusters:
-        found_clusters = len(np.unique(self.defence.clusters_by_class[0]))
-        found_clusters_gen = len(np.unique(self.defence_gen.clusters_by_class[0]))
-        self.assertEqual(found_clusters, 3)
-        self.assertEqual(found_clusters_gen, 3)
-
-        # Test clean data has changed
-        sum_clean2 = sum(is_clean_lst)
-        sum_clean2_gen = sum(is_clean_lst_gen)
-        self.assertNotEqual(sum_clean1, sum_clean2)
-        self.assertNotEqual(sum_clean1_gen, sum_clean2_gen)
-
-        kwargs = {"nb_clusters": 2, "nb_dims": 10, "reduce": "PCA", "cluster_analysis": "distance"}
-        _, is_clean_lst = self.defence.detect_poison(**kwargs)
-        _, is_clean_lst_gen = self.defence_gen.detect_poison(**kwargs)
-        sum_dist = sum(is_clean_lst)
-        sum_dist_gen = sum(is_clean_lst_gen)
-        kwargs = {"nb_clusters": 2, "nb_dims": 10, "reduce": "PCA", "cluster_analysis": "smaller"}
-        _, is_clean_lst = self.defence.detect_poison(**kwargs)
-        _, is_clean_lst_gen = self.defence_gen.detect_poison(**kwargs)
-        sum_size = sum(is_clean_lst)
-        sum_size_gen = sum(is_clean_lst_gen)
-        self.assertNotEqual(sum_dist, sum_size)
-        self.assertNotEqual(sum_dist_gen, sum_size_gen)
-
-    def test_evaluate_defense(self):
-        # Get MNIST
-        (x_train, _), (_, _), (_, _) = self.mnist
-
-        kwargs = {"nb_clusters": 2, "nb_dims": 10, "reduce": "PCA"}
-        _, _ = self.defence.detect_poison(**kwargs)
-        _, _ = self.defence_gen.detect_poison(**kwargs)
-        is_clean = np.zeros(len(x_train))
-        self.defence.evaluate_defence(is_clean)
-        self.defence_gen.evaluate_defence(is_clean)
-
-    def test_analyze_cluster(self):
-        # Get MNIST
-        (x_train, _), (_, _), (_, _) = self.mnist
-
-        self.defence.analyze_clusters(cluster_analysis="relative-size")
-        self.defence_gen.analyze_clusters(cluster_analysis="relative-size")
-
-        self.defence.analyze_clusters(cluster_analysis="silhouette-scores")
-        self.defence_gen.analyze_clusters(cluster_analysis="silhouette-scores")
-
-        report, dist_clean_by_class = self.defence.analyze_clusters(cluster_analysis="distance")
-        report_gen, dist_clean_by_class_gen = self.defence_gen.analyze_clusters(cluster_analysis="distance")
-        n_classes = self.classifier.nb_classes
-        self.assertEqual(n_classes, len(dist_clean_by_class))
-        self.assertEqual(n_classes, len(dist_clean_by_class_gen))
-
+@pytest.fixture()
+def get_ac(get_default_mnist_subset):
+    (x_train, y_train), (x_test, y_test), min_, max_ = load_mnist()
```

Comment:
Can we replace `load_mnist` with `get_default_mnist_subset`? The min and max are constant for `get_default_mnist_subset`.

Reply:
@ebubae Did you notice this comment?